recipe/patches/0001-remove-modules-needing-pythonservice.cpp-for-pypy.patch

```diff
@@ -0,0 +1,74 @@
+From 473a35adafc1e9a13d8a5c76bfbbb1d0c8a0cb3d Mon Sep 17 00:00:00 2001
+From: Isuru Fernando <isuruf@gmail.com>
+Date: Mon, 23 Aug 2021 07:08:14 -0500
+Subject: [PATCH] remove modules needing pythonservice.cpp for pypy
+
+---
+ setup.py | 30 +++++++++++++++++++-----------
+ 1 file changed, 19 insertions(+), 11 deletions(-)
+
+diff --git a/setup.py b/setup.py
+index 1d0d0a8d..0485383c 100644
+--- a/setup.py
++++ b/setup.py
+@@ -113,6 +113,8 @@ try:
+ except ValueError:
+     skip_verstamp = False
+ 
++IS_PYPY = sys.implementation.name == 'pypy'
++
+ try:
+     this_file = __file__
+ except NameError:
+@@ -1488,13 +1490,15 @@ win32_extensions += [
+            windows_h_version=0x0500,
+         ),
+ ]
+-win32_extensions += [
+-    WinExt_win32('servicemanager',
+-           sources = ["win32/src/PythonServiceMessages.mc", "win32/src/PythonService.cpp"],
+-           extra_compile_args = ['-DPYSERVICE_BUILD_DLL'],
+-           libraries = "user32 ole32 advapi32 shell32",
+-           windows_h_version = 0x500),
+-]
++
++if not IS_PYPY:
++    win32_extensions += [
++        WinExt_win32('servicemanager',
++            sources = ["win32/src/PythonServiceMessages.mc", "win32/src/PythonService.cpp"],
++            extra_compile_args = ['-DPYSERVICE_BUILD_DLL'],
++            libraries = "user32 ole32 advapi32 shell32",
++            windows_h_version = 0x500),
++    ]
+ 
+ win32_extensions += [
+     WinExt_win32('win32help',
+@@ -2076,10 +2080,6 @@ other_extensions.append(
+ )
+ 
+ W32_exe_files = [
+-    WinExt_win32_subsys_con("pythonservice",
+-         sources=[os.path.join("win32", "src", s) for s in
+-                  "PythonService.cpp PythonService.rc".split()],
+-         libraries = "user32 advapi32 ole32 shell32"),
+     WinExt_pythonwin_subsys_win("Pythonwin",
+         sources = [
+             "Pythonwin/pythonwin.cpp",
+@@ -2093,6 +2093,14 @@ W32_exe_files = [
+         optional_headers=['afxres.h']),
+ ]
+ 
++if not IS_PYPY:
++    W32_exe_files += [
++        WinExt_win32_subsys_con("pythonservice",
++            sources=[os.path.join("win32", "src", s) for s in
++                  "PythonService.cpp PythonService.rc".split()],
++            libraries = "user32 advapi32 ole32 shell32"),
++    ]
++
+ # Special definitions for SWIG.
+ swig_interface_parents = {
+     # source file base,     "base class" for generated COM support
+-- 
+2.28.0
+
```

Comment:
I think it might be cleaner to do something like this
```
diff --git a/setup.py b/setup.py
index 297d15cb..40794da1 100644
--- a/setup.py
+++ b/setup.py
@@ -102,6 +102,8 @@ try:
 except KeyError:
     pass # no EXT_SUFFIX in this build.
 
+IS_PYPY = sys.implementation.name == "pypy"
+
 build_id_patch = build_id
 if not "." in build_id_patch:
     build_id_patch = build_id_patch + ".0"
@@ -635,6 +637,10 @@ class my_build_ext(build_ext):
         if ext.platforms and self.plat_name not in ext.platforms:
             return "Only available on platforms %s" % (ext.platforms,)
 
+        if IS_PYPY and ext.name in ('servicemanager', 'pythonservice',
+                                    'pythoncom', 'PyISAPI_loader'):
+            return "Cannot build service extensions on PyPy"
+
         # We update the .libraries list with the resolved library name.
         # This is really only so "_d" works.
         ext.libraries = patched_libs
```

Reply:
Sounds good. Can you push your changes here?

Reply:
I am working out exactly which modules fail locally. So far I need to add 'adsi' to this list, maybe more.